Rubberduck.CodeAnalysis/CodePathAnalysis/Walker.cs

```diff
@@ -13,33 +14,68 @@ public class Walker
         public INode GenerateTree(IParseTree tree, Declaration declaration)
         {
             AssignmentNode lastAssignment = null;
-            return GenerateTree(tree, declaration, ref lastAssignment);
+            bool isBranchCondition = false;
+            return GenerateTree(tree, declaration, ref lastAssignment, ref isBranchCondition);
         }
 
-        public INode GenerateTree(IParseTree tree, Declaration declaration, ref AssignmentNode lastAssignment, bool isConditional = false, bool isInsideLoop = false)
+        public INode GenerateTree(IParseTree tree, Declaration declaration, ref AssignmentNode lastAssignment, ref bool isBranchCondition)
         {
             INode node = default;
+            VBAParser.BooleanExpressionContext branchCondition;
             switch (tree)
             {
                 case VBAParser.ForNextStmtContext _:
                 case VBAParser.ForEachStmtContext _:
                 case VBAParser.WhileWendStmtContext _:
                 case VBAParser.DoLoopStmtContext _:
                     node = new LoopNode(tree);
-                    isInsideLoop = true;
                     break;
+
                 case VBAParser.IfStmtContext _:
+                    node = new BranchNode(tree);
+                    isBranchCondition = true;
+                    break;
                 case VBAParser.ElseBlockContext _:
+                    node = new BranchNode(tree);
+                    lastAssignment = null;
```

Comment:
Is it really a concern of a code path analysis? This seems to be more specialized than the type suggests. When code path analysis API matures, this might need to be refactored into more specialized type. That said, I'm fine with merging it as-is as long it has a `TODO` marker to note that.

Reply:
The `lastAssignment` is an `AssignmentNode` whose `Usages` grow with each `ReferenceNode` objects as the tree is traversed - that said I need to add more tests, not sure this state reset for `Else` blocks is covered yet.